src/commands/trace/trace.ts

```diff
@@ -0,0 +1,121 @@
+import chalk from 'chalk'
+import {spawnSync} from 'child_process'
+import {Command} from 'clipanion'
+import crypto from 'crypto'
+import os from 'os'
+import {performance} from 'perf_hooks'
+import {parseTags} from '../../helpers/tags'
+import {apiConstructor} from './api'
+import {APIHelper} from './interfaces'
+
+export class TraceCommand extends Command {
+  public static usage = Command.Usage({
+    description: 'Trace a command with a custom span and report it to Datadog.',
+    details: `
+            This command wraps another command, which it will launch, and report a custom span to Datadog.
+            See README for details.
+        `,
+    examples: [
+      ['Trace a command and report to Datadog', 'datadog-ci trace echo "Hello World"'],
```

Comment:
I'm not sure if this is helpful, but .NET does something like this:
```
datadog-ci trace -- echo "Hello World"
```
https://docs.datadoghq.com/continuous_integration/setup_tests/dotnet/#instrumenting-tests 

I believe the idea being that we can more easily tell arguments apart, e.g.
```
datadog-ci trace --service=my-service -- echo --help
```

Reply:
So I did start with that syntax, but I got a message from yarn saying that `--` was deprecated or something like that. I'll add it back.